src/test/java/sp/SecondPartTasksTest.java

```diff
@@ -0,0 +1,66 @@
+package sp;
+
+import com.google.common.collect.ImmutableMap;
+import org.junit.Test;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.*;
+import static sp.SecondPartTasks.*;
+
+public class SecondPartTasksTest {
+
+    @Test
+    public void testFindQuotes() {
+        final String file1 = "src/test/resources/Californication.txt";
+        final String file2 = "src/test/resources/Sweet Dreams.txt";
+
+        assertEquals(findQuotes(Collections.emptyList(), ""), Collections.emptyList());
+        assertEquals(findQuotes(Arrays.asList(file1), "Dream").stream().distinct().collect(Collectors.toList()),
+                Arrays.asList("Dream of silver screen quotations", "Dream of Californication"));
+        assertEquals(findQuotes(Arrays.asList(file1), "dreams").stream().distinct().collect(Collectors.toList()),
+                Arrays.asList("And if you want these kind of dreams"));
+        assertEquals(findQuotes(Arrays.asList(file2), "dreams").stream().distinct().collect(Collectors.toList()),
+                Arrays.asList("Sweet dreams are made of this"));
+        assertEquals(findQuotes(Arrays.asList(file1, file2), "dreams").stream().distinct().collect(Collectors.toList()),
+                Arrays.asList("And if you want these kind of dreams", "Sweet dreams are made of this"));
+        assertEquals(findQuotes(Arrays.asList(file1, file2), "Dream").stream().distinct().collect(Collectors.toList()),
+                Arrays.asList("Dream of silver screen quotations", "Dream of Californication"));
```

Comment:
asEq(new HashSet(findQuotes(Arrays.asList(file1, file2), "Dream")), ImmutableMap.of(...))